### PR TITLE
Added separate crypto module to Tezos_interop

### PR DESCRIPTION
**Pull request by Zett98**
## Problem

We need to support tz3 addresses which uses p256.

## Solution

Added separate crypto module with p256 and ed25519 support to tezos_interop.

### CODE REVIEW

File: tezos_interop/crypto.rei [hunk 1]
```diff
@@ -0,0 +1,43 @@
+open Helpers;
+open Mirage_crypto_ec;
+
+let blake2b_20_encoding: Data_encoding.t(BLAKE2B_20.t);
+module type S = {
```
Comment (EduardoRFS) [inline tezos_interop/crypto.rei:5]: Any tip to avoid duplicating this module type?

**Review by EduardoRFS — CHANGES_REQUESTED** [review]
Seems like this module is not used at all.

File: tezos_interop/crypto.rei [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  module Pub: {
```
[the remark this answers is not in the document]
Comment (Zett98) [reply tezos_interop/crypto.rei:6]: i looked at this https://gitlab.com/tezos/tezos/-/blob/master/src/lib_crypto/p256.ml and ed25519.ml

File: tezos_interop/crypto.re [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      bytes =>
+        Bytes.to_string(bytes) |> BLAKE2B_20.of_raw_string |> Option.get,
+      Fixed.bytes(20),
+    )
+  );
+module type S = {
+  module Pub: {
+    type t;
+    let encoding: Data_encoding.t(t);
+    let to_string: (~alphabet: Base58.Alphabet.t=?, t) => string;
```
[the remark this answers is not in the document]
Comment (Zett98) [reply tezos_interop/crypto.re:18]: fixed

File: tezos_interop/crypto.rei (hunk 1 above)
Comment (Zett98) [reply tezos_interop/crypto.rei:5]: used an intf file

**Comment by Zett98** [comment]
> 
> 
> Seems like this module is not used at all.

i wanted to make this pull request smaller, but now i used the module in ```tezos_interop.re```

**Review by anchpop — APPROVED** [review]

File: tezos_interop/crypto.rei (hunk 2 above)
[the remark this answers is not in the document]
Comment (EduardoRFS) [reply tezos_interop/crypto.rei]: Yeah they're internally referred in a couple different ways which is just bad, but externally at michelson you can find it being called `key`.

**Review by callistonianembrace — APPROVED** [review]

File: tezos_interop/crypto.re [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      let ident = 2 + Cstruct.get_uint8(b, 63) land 1;
+      Cstruct.blit(b, 1, res, 1, 32);
+      Cstruct.set_uint8(res, 0, ident);
+      repr(res);
+    };
+    let compress_to_string = compress(Cstruct.to_string);
+    let compress_to_bytes = compress(Cstruct.to_bytes);
+    let p =
+      Z.of_string(
+        "115792089210356248762697446949407573530086143415290314195533631308867097853951",
```
[the remark this answers is not in the document]
Comment (Zett98) [reply tezos_interop/crypto.re:118]: documented domain params of the curve.

> Also can you open a PR on mirage-crypto-ec with this?

Is this is about compression? (i thought about contributing it after i finish up with tz2 and tz3 support)
Or is it about the numbers in which case you can find them in HEX form inside the mirage-crypto repo
b: https://github.com/mirage/mirage-crypto/blob/main/ec/mirage_crypto_ec.ml#L639
p: https://github.com/mirage/mirage-crypto/blob/main/ec/mirage_crypto_ec.ml#L644

**Pull request closed without merging**